src/olympia/translations/models.py

```diff
@@ -249,34 +243,25 @@ class PurifiedMarkdownTranslation(PurifiedTranslation):
     class Meta:
         proxy = True
 
-    def clean_localized_string(self):
-        # bleach user-inputted html
+    def clean_string(self, text):
+        # bleach user-inputted html first.
         cleaned = (
-            bleach.clean(self.localized_string, tags=[], attributes={})
-            if self.localized_string
-            else ''
+            nh3.clean(text, tags=set(), attributes={}) if self.localized_string else ''
         )
         # hack; cleaning breaks blockquotes
         text_with_brs = cleaned.replace('&gt;', '>')
         # the base syntax of markdown library does not provide abbreviations or fenced
         # code. see https://python-markdown.github.io/extensions/
-        markdown = md.markdown(text_with_brs, extensions=['abbr', 'fenced_code'])
-
-        # Keep only the allowed tags and attributes, strip the rest.
-        cleaner = bleach.Cleaner(
-            tags=self.allowed_tags,
-            attributes=self.allowed_attributes,
-            filters=[self.linkify_filter],
-            strip=True,
-        )
+        html = md.markdown(text_with_brs, extensions=['abbr', 'fenced_code'])
 
-        return cleaner.clean(markdown)
+        # Run through cleaning as normal
+        return super().clean_string(html)
```

Comment:
This is causing a subtle breakage: We used to get text with `<p>` elements inserted by `md.markdown()`, only to remove them with `bleach` (with `strip=True`) and have them replaced with newlines, which got the string back to the original number of newlines.

We are now running through regular cleaning with `nh3` which strips but doesn't "restore" the newlines... we can work around that, maybe converting `\n` to `br` early and allowing `<br>` after markdown, but need to be careful.